rxgroups/src/main/java/com/airbnb/rxgroups/ManagedObservable.java

```diff
@@ -86,25 +86,6 @@ String tag() {
     return tag;
   }
 
-  @Override public boolean equals(Object o) {
```

Comment:
I spent a couple hours debugging this and turns out that the current implementation is broken.
I'm not sure why we decided to override equals/hashCode in the first place, but I noticed that when `onDestroy()` was called, it was unsubscribing from the group, which was causing the `hashCode` to change (because we're setting the observer to null), so the `if (hashCode == manager.hashCode())` check was always returning false :(